### PR TITLE
How to create a hook

### LINKED ISSUE #11: create a HOW-CREATE-A-HOOK.md
We also need a guide to show how the developers can creating new hooks.

### PR DESCRIPTION
**Pull request by elainemattos**
Hey!

This PR is for documentation of how to create a hook!

Please let me know if there's anything else to be amended.

Closes #11

### CODE REVIEW

**Review by tauantcamargo — APPROVED** [review]

**Comment by tauantcamargo** [comment]
@elainemattos nice job, just a thing, can u add a link to HOW-CREATE-A-HOOK.md in README.md before Contributing link :smile: ?

image

File: README.md [hunk 1]
```diff
@@ -94,6 +94,7 @@ export default Carousel
 
 ## Documentation
 
+- [How to Create a Hook](HOW-CREATE-A-HOOK.md)
```
Comment (talyssonoc) [inline README.md:97]: I think this should be linked in https://github.com/devthiago/react-ui-hooks/blob/master/CONTRIBUTING.md as well as part of the `Make the change` section. What do you think?

Comment (tauantcamargo) [reply README.md:97]: i think this could appear in both, what do you think? @talyssonoc @elainemattos @devthiago

File: HOW-CREATE-A-HOOK.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+#### Create a Hook file
+
+The `generate:hook` command create 3 files:
+
+- A hook file ***useExample.js***
+- The tests file ***useExample.test.js*** 
+- The documentation file ***useExample.md***
+
+The hook and tests files must to be updated manually, the documentation file can be updated by the `generate:docs` command all at once.
```
Comment (gutofoletto) [inline HOW-CREATE-A-HOOK.md]:
```suggestion
The hook and tests files must be updated manually, the documentation file can be updated all at once by using the `generate:docs` command .
```

File: HOW-CREATE-A-HOOK.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+***useExample.test.js*** on src/tests/hooks 
+
+***useExample.md*** on docs/
+
+------------
+
+### Create a new react hook unit test
+
+If you've created your hook manually you can generate the unit test file for it with the command:
```
Comment (gutofoletto) [inline HOW-CREATE-A-HOOK.md]:
```suggestion
If you've created your hook manually you can generate its unit test file with the command:
```

File: HOW-CREATE-A-HOOK.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+```
+or
+```bash
+yarn generate:docs useExample.js
+```
+------------
+
+### Generate docs for all react hooks:
+
+As you update the hooks and tests frequently there's a need to to update the documentation, so you can do it for all of the hooks with:
```
Comment (gutofoletto) [inline HOW-CREATE-A-HOOK.md]:
```suggestion
As you update the hooks and tests frequently there's a need to update the documentation, so you can do it for all of the hooks with:
```

File: HOW-CREATE-A-HOOK.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+```bash
+npm run generate:docs
+```
+or
+
+```bash
+yarn generate:docs
+```
+
+This command is going to check all of your hooks and hooks tests and generate the documentation!
```
Comment (gutofoletto) [inline HOW-CREATE-A-HOOK.md]:
```suggestion
This command scans your hooks and their tests to generate the documentation.
```

**Review by gutofoletto — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (talyssonoc) [reply README.md:97]: That's what I mean by "as well" 😛 , I agree!

File: HOW-CREATE-A-HOOK.md (hunk 5 above)
Comment (talyssonoc) [reply HOW-CREATE-A-HOOK.md]: @gutofoletto I guess it's "scans your hooks and _their_ tests"

File: README.md (hunk 1 above)
Comment (tauantcamargo) [reply README.md:97]: i'm sorry, i was thinking that you were talking to get this out of the readme :(

**Review by tauantcamargo — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (elainemattos) [reply README.md:97]: at CONTRIBUTING.md
[...] 
- Make the tests.
- Check if the tests are all passing `yarn test`.
***Detailed explanation can be found at [How to Create a Hook](HOW-CREATE-A-HOOK.md)***

What do you think?

Comment (tauantcamargo) [reply README.md:97]: @elainemattos at CONTRIBUTING.md i think you can change this

image

to something like that =>

### Create new approaches or make changes

- Make the changes or create a new Hook (Detailed explanation can be found at [How to Create a Hook]())

What do you think?

Comment (elainemattos) [reply README.md:97]: Seems better, I'll make the changes

**Review by tauantcamargo — APPROVED** [review]

**Review by hjJunior — APPROVED** [review]

File: HOW-CREATE-A-HOOK.md (hunk 5 above)
Comment (gutofoletto) [reply HOW-CREATE-A-HOOK.md]: @talyssonoc indeed! my bad :P

**Review by devthiago — APPROVED** [review]
LGTM 👍

**Comment by gutofoletto** [comment]
Closes #11

**Pull request merged**